framework/python/src/api/api.py

```diff
@@ -243,8 +247,35 @@ async def stop_test_run(self):
   async def get_status(self):
     return self._test_run.get_session().to_json()
 
+  def shutdown(self, response: Response):
+
+    LOGGER.debug("Received request to shutdown Testrun")
+
+    # Check that Testrun is not currently running
+    if (self._session.get_status() not in [
+            "Cancelled",
+            "Compliant",
+            "Non-Compliant",
+            "Idle"]):
+      LOGGER.debug("Unable to shutdown Testrun as Testrun is in progress")
+      response.status_code = 400
+      return self._generate_msg(
+        False,
+        "Unable to shutdown. A test is currently in progress."
+      )
+
+    self._test_run.shutdown()
+    os.kill(os.getpid(), signal.SIGTERM)
+
   async def get_version(self, response: Response):
+
+    # Add defaults
     json_response = {}
+    json_response["installed_version"] = "v1.0"

```

Comment:
Hard coding doesn't give correct results.  Should resolve current version from the session:

`json_response["installed_version"] = self._test_run.get_session().get_version()`

Reply:
This is done shortly after in the code. The idea here was to add a default fallback value in case resolving the current version fails for whatever reason.

Reply:
If that's the case, we need  report some sort of unresolved version if this fails or needs a different reliable way to have a default installed version reported.  Hard coding a wrong version doesn't solve this, just opens the door for bad version reporting.